src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -149,18 +160,26 @@ public void setupChild(Configuration childConf)
             throws CheckstyleException {
         final String name = childConf.getName();
         final Object module = moduleFactory.createModule(name);
-        if (!(module instanceof AbstractCheck)) {
+        if (module instanceof AbstractCheck || module instanceof TreeWalkerFilter) {
+            final AutomaticBean bean = (AutomaticBean) module;
+            bean.contextualize(childContext);
+            bean.configure(childConf);
+            if (module instanceof AbstractCheck) {
+                final AbstractCheck check = (AbstractCheck) module;
+                check.init();
+                registerCheck(check);
+            }
+            else {
+                final TreeWalkerFilter filter = (TreeWalkerFilter) module;
+                filters.add(filter);
+            }
+        }
+        else {
             throw new CheckstyleException(
-                "TreeWalker is not allowed as a parent of " + name
-                        + " Please review 'Parent Module' section for this Check in web"
-                        + " documentation if Check is standard.");
+                    "TreeWalker is not allowed as a parent of " + name
```

Comment:
@timurt 
Do not ignore [this point](https://github.com/checkstyle/checkstyle/pull/4755/files/e42da8b7f713e5eb3ccb3ff63a8351284e234575#r128892154)